### DISCUSSION TITLE
How to configure apache CXF property org.apache.cxf.stax.maxChildElements in quarkus

### DISCUSSION THREAD

**Discussion by hikemachado** [quarkiverse/quarkus-cxf#1325] [Q&A] [answered]
Hello guys,

How could I configure org.apache.cxf.stax.maxChildElements in quarkus?
I added org.apache.cxf.stax.maxChildElements=100000 in the application.properties but nothing happened

I'm trying to solve the exception below:

jakarta.xml.ws.soap.SOAPFaultException: Unmarshalling Error: Maximum Number of Child Elements limit (50000) Exceeded 

Thanks
HM

**Comment by ppalaga** [comment, marked as answer]
As of Quarkus CXF 3.9.0, there is no way to do it via `application.properties`.

Around [here](https://cxf.apache.org/docs/security.html#Security-ControllingLargeRequestPayloads) the CXF docs says it can be done per Bus or Endpoint. If Bus is fine for you, an Init bean like this should work:

```java
import jakarta.enterprise.event.Observes;

import org.apache.cxf.Bus;
import org.apache.cxf.BusFactory;

import io.quarkus.runtime.StartupEvent;

public class Init {

    void onStart(@Observes StartupEvent ev) {
        final Bus bus = BusFactory.getDefaultBus();
        bus.setProperty("org.apache.cxf.stax.maxChildElements", "1024");

    }
}
```

**Reply by ppalaga** [reply]
Please [file an issue](https://github.com/quarkiverse/quarkus-cxf/issues) if you think configuring this via `application.properties` is a good idea.

**Reply by hikemachado** [reply]
Thanks a lot,
It solved the problem , I think it is not necessary to file an issue as I can add a @ConfigProperty in  that class and set the property in the applications.properties as well

Thanks
